### PR TITLE
8327253: [lworld] Preloading of classes needs a complete rework

### PR DESCRIPTION
**Pull request by fparain**
Rework of class preloading to fix existing issues and support new preloading scheme (see CR for details).
Also change the array of inline types fields classes from InlineKlass** to Array<InlineKlass*>*, much easier to manage.

Fred

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8327253](https://bugs.openjdk.org/browse/JDK-8327253): [lworld] Preloading of classes needs a complete rework (**Bug** - P2)

### Reviewers
 * [David Simms](https://openjdk.org/census#dsimms) (@MrSimms - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/999/head:pull/999` \
`$ git checkout pull/999`

Update a local copy of the PR: \
`$ git checkout pull/999` \
`$ git pull https://git.openjdk.org/valhalla.git pull/999/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 999`

View PR using the GUI difftool: \
`$ git pr show -t 999`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/999.diff">https://git.openjdk.org/valhalla/pull/999.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/999#issuecomment-1977509443)

### CODE REVIEW

File: src/hotspot/cpu/x86/macroAssembler_x86.cpp
Comment (MrSimms) [inline src/hotspot/cpu/x86/macroAssembler_x86.cpp:1]: Fairly minor change for aarch64, could we include that too ?

Comment (fparain) [reply src/hotspot/cpu/x86/macroAssembler_x86.cpp:1]: Good catch! Fixed.

**Review by MrSimms — APPROVED** [review]
LGTM !

**Comment by fparain** [comment]
Thanks for the reviews.

**Pull request closed without merging**